### ISSUE TITLE
TextFields values not accessible in iOS7?

### ISSUE THREAD

**Issue by redshoulderedhawk** [sberrevoets/SDCAlertView#67]
Hello,

I'm really liking SDCAlertViewController, but I'm seeing a funny difference between iOS7 and iOS8 with regards to UITextFields.  In iOS8, I can use the code below to easily access the value of a text field in a hander, but in iOS7.1 the same code gives a blank text instead (not nil).  Am I doing this correct?

Thanks!

```
[self.addAlert addAction:[SDCAlertAction actionWithTitle:@"Add" style:SDCAlertActionStyleDefault handler:^(SDCAlertAction *action) {
    NSString *name = ((UITextField *) [self.addAlert.textFields objectAtIndex:0]).text;
    [self somefunction:name];
}]];
```

**Comment by castles** [comment]
Has there been any updates to this? I'm having issues setting and retrieving the textfield value in is. 

Using the code above I'm getting an error saying "Receiver type 'SDCAlertView' for instance message is a forward declaration"

**Comment by sberrevoets** [comment]
No actual fix yet, sorry. However, you can most likely solve your problem by adding a `#import "SDCAlertView.h" statement at the top of your file.

**Comment by castles** [comment]
ahh yes, thanks. That fixed it.

**Comment by redshoulderedhawk** [comment]
There was a slight typo in my example code above, I had put a 0 instead of textFieldIndex ... So I'm copying in the full code for an Objective-C Category.  It is called "SDCAlertController+UITextField".  Since, this is implemented as a category, updating to the latest code shouldn't break any code that uses the work around.  Hopefully this is helpful to someone. 

Here it is.

SDCAlertController+UITextField.h:
```
#import <Foundation/Foundation.h>
#import "SDCAlertController.h"


@interface SDCAlertController(UITextField)
- (UITextField *)textFieldAtIndex:(NSInteger)textFieldIndex;
@end
```

SDCAlertController+UITextField.m
```
#import "SDCAlertController+UITextField.h"
#import "SDCAlertView.h"

@implementation SDCAlertController(UITextField)

- (UITextField *)textFieldAtIndex:(NSInteger)textFieldIndex {
    
    if (self.legacyAlertView) {
        return [self.legacyAlertView textFieldAtIndex:textFieldIndex];
    } else {
        return [self.textFields objectAtIndex:textFieldIndex];
    }
}

@end
```

**Comment by sberrevoets** [comment]
I have not found a convenient way to make the compatibility better, so if you want to submit a PR with that code in it, I'll merge it in so that others can use it as well.

If you do submit a PR, I ask that you add this method to the `Legacy` category in SDCAlertController.h so that when support for `SDCAlertView` is eventually dropped, I can just remove that entire category.

Thanks for your follow-ups, @redshoulderedhawk!

**Comment by redshoulderedhawk** [comment]
See the PR that I just created.  Thanks for the project!

**Comment by sberrevoets** [comment]
Just merged #78
